app/services/authentication/providers/github.rb

```diff
@@ -9,9 +9,10 @@ class Github < Provider
 
       def new_user_data
         name = raw_info.name.presence || info.name
+        email = auth_payload.extra.all_emails.detect(&:primary).email
```

Comment:
I think this doesn't work as `all_emails` is an array of hashes whose `primary` key is a string. BTW as far as I know `.info` always contains the primary email if there is one. 

See the implementation here 

https://github.com/omniauth/omniauth-github/blob/d001ed274276bf50e1e756bf38bce75509aa25ec/lib/omniauth/strategies/github.rb#L58-L61

and

https://github.com/omniauth/omniauth-github/blob/d001ed274276bf50e1e756bf38bce75509aa25ec/lib/omniauth/strategies/github.rb#L50-L52

and 

https://github.com/omniauth/omniauth-github/blob/d001ed274276bf50e1e756bf38bce75509aa25ec/lib/omniauth/strategies/github.rb#L28-L39

Reply:
Hmm I think it does work because the array is a special array, `Hashie::Array`:
![Screen Shot 2020-08-10 at 12 48 15 PM](https://user-images.githubusercontent.com/17884966/89808199-c7b4f280-db07-11ea-9c3c-3cc91f8a260b.png)

That said my implementation doesn't cover for `verified` which the omniauth gem does. I think I know the problem now though!

Reply:
Oh interesting! Sorry I grabbed the JSON payload without sending it through Hashie :D